packages/server/lib/browsers/chrome.ts

```diff
@@ -519,40 +503,46 @@ export = {
     // first allows us to connect the remote interface,
     // start video recording and then
     // we will load the actual page
-    const launchedBrowser = await launch(browser, 'about:blank', port, args)
+    const launchedBrowser = await launch(browser, 'about:blank', port, args) as LaunchedBrowser & { browserCriClient: BrowserCriClient}
 
     la(launchedBrowser, 'did not get launched browser instance')
 
     // SECOND connect to the Chrome remote interface
     // and when the connection is ready
     // navigate to the actual url
-    const criClient = await this._connectToChromeRemoteInterface(port, options.onError, browser.displayName)
+    const browserCriClient = await BrowserCriClient.create(port, browser.displayName, options.onError)
 
-    la(criClient, 'expected Chrome remote interface reference', criClient)
+    la(browserCriClient, 'expected Chrome remote interface reference', browserCriClient)
 
-    await criClient.ensureMinimumProtocolVersion('1.3')
+    await browserCriClient.ensureMinimumProtocolVersion('1.3')
     .catch((err) => {
       // if this minumum chrome version changes, sync it with
       // packages/web-config/webpack.config.base.ts and
       // npm/webpack-batteries-included-preprocessor/index.js
       throw new Error(`Cypress requires at least Chrome 64.\n\nDetails:\n${err.message}`)
     })
 
-    this._setAutomation(criClient, automation)
-
     // monkey-patch the .kill method to that the CDP connection is closed
     const originalBrowserKill = launchedBrowser.kill
 
+    launchedBrowser.browserCriClient = browserCriClient
```

Comment:
Should (launch) accept the browserCriClient as a parameter so we aren't attaching like this? Or should launch be creating the criClient then we interfact with it through launchBrowser?

```js
const launchedBrowser = await launch(browser, 'about:blank', port, args, options.onError)
la(launchedBrowser.browserCriClient, 'expected Chrome remote interface reference', browserCriClient)

awaitl aunchedBrowser.browserCriClient.ensureMinimumProtocolVersion('1.3')
```

Reply:
It's sort of a chicken and the egg problem here. We can't have the browserCriClient until we launch the browser. I could add the browserCriClient connection process to launch, but that package is very scoped to just starting processes so I want to avoid that. I could also have browserCriClient be a singleton in this file, or even have a mapping of process ids to browserCriClients

Reply:
I ended up going with the mapping option: https://github.com/cypress-io/cypress/commit/6c2f7fb16de44eb9aead83e3b8e6f8778f78d1b2